frontend/src/api/actions/init.js

```diff
@@ -0,0 +1,119 @@
+import { setActiveSession, fetchSessions, sessionsSelector } from "./sessions";
+import { fetchActiveUser, setActiveUserRole } from "./users";
+import { fetchApplicants } from "./applicants";
+import { fetchApplications } from "./applications";
+import { fetchAssignments } from "./assignments";
+import { fetchContractTemplates } from "./contract_templates";
+import { fetchInstructors } from "./instructors";
+import { fetchPositions } from "./positions";
+import { setGlobals, globalsSelector } from "./globals";
+
+function toggleMockApi(enableMockAPI) {
+    // in production, ToggleMockApi is a no-op. In development, it actually
+    // does something.
+    /* eslint-disable */
+    if (enableMockAPI != null && process.env.REACT_APP_DEV_FEATURES) {
+        const mock = require("../mockAPI");
+        const mockAPI = mock.mockAPI;
+
+        if (enableMockAPI === true) {
+            mockAPI.replaceGlobalFetch();
+        } else {
+            mockAPI.restoreGlobalFetch();
+        }
+    }
+    /* eslint-enable */
+}
+
+function updateGlobals(enableMockAPI, newActiveSession, getState) {
+    const state = getState();
+    let globals = globalsSelector(state);
+
+    if (enableMockAPI != null) {
+        globals = {
+            ...globals,
+            mockAPI: enableMockAPI
+        };
+    }
+
+    if (newActiveSession) {
+        globals = {
+            ...globals,
+            activeSession: newActiveSession.id
+        };
+    }
+
+    return setGlobals(globals);
+}
+
+export function initFromStage(stage, options) {
+    return async (dispatch, getState) => {
+        const onInitialization = [
+            "toggleMockAPI",
+            "setActiveUser",
+            "setActiveUserRole",
+            "fetchSessions",
+            "setActiveSession",
+            "updateGlobals"
+        ];
+
+        const {
+            activeSession = null,
+            activeUserRole = "admin",
+            mockAPI = null
+        } = options;
+
+        const startIndex = onInitialization.indexOf(stage);
+
+        if (startIndex <= onInitialization.indexOf("toggleMockAPI")) {
```

Comment:
Instead of `startIndex` and testing it each time, make a `shouldRunStage(<string>) -><bool>` function that does the test. That should make the code intention a bit clearer.